Azure/azure-cli
# [Core] New error output

**Description**

This PR applies the new error output to support the feature requests #15693.  

**1. The UI changes of new error output includes** 
- the new output UI designs
- removing error types
- multiple recommended commands (at most 3 commands + 1 reference link)
- descriptions of the recommended commands
- colorizations
- command group level hyperlinks

An overview of the new error output.

![image](https://user-images.githubusercontent.com/62928370/102180544-01e23680-3ee4-11eb-87ea-416acc1ff488.png)


**2. The none-UI changes**
- Set `core.disable_error_recommendation=true` with az config to disable recommendations
- The recommended commands are re-ordered by parameter matching with user's input
- The recommended commands are eithe all from Aladdin service or all from examples, in avoid of duplicate commands
- Contextual values are disabled to replace the placeholders as it will make command description not valid. 

**Testing Guide**

Type any command that will triger the client side userfault. E.g, the command below, you shold see the output above.
```
az vm create -n test-vm
```

Disable the error recommendation and then test the same command, you could see the recommendations are disabled.
```
az config set core.error_recommendation=off
az vm create -n test-vm
```
![image](https://user-images.githubusercontent.com/62928370/106561446-a6d3d400-6563-11eb-9fd9-31359bad5f9b.png)
